client/src/context/reducer.js

```diff
@@ -1,44 +1,47 @@
+/** @format */
+
 export const initialState = {
-  authenticated: false,
-  user: {},
-  cart: { flights: [], hotels: [], rentalCar: [] },
+	authenticated: false,
+	user: {},
+	cart: { flights: [], hotels: [], rentalCar: [] },
 };
 
 export const userReducer = (state = initialState, action) => {
-  console.log(action);
-  switch (action.type) {
-    case "AUTHENTICATED":
-      return {
-        ...state,
-        authenticated: true,
-        user: action.payload.user,
-      };
-    case "UNAUTHENTICATED":
-      return {
-        ...state,
-        authenticated: false,
-        user: null,
-      };
-    case "ADD_TO_CART":
-      return {
-        ...state,
-        cart: {
-          ...state.cart,
-          [action.purhcaseType]: [
-            ...state.cart[action.purhcaseType],
-            action.item,
-          ],
-        },
-      };
-    case "REMOVE_FROM_CART":
-      return {
-        ...state,
-        cart: {
-          ...state.cart,
-          [action.purchaseType]: [],
-        },
-      };
-    default:
-      throw new Error(`Unhandled action type: ${action.type}`);
-  }
+	switch (action.type) {
+		case "AUTHENTICATED":
+			return {
+				...state,
+				authenticated: true,
+				user: action.payload.user,
+			};
+		case "UNAUTHENTICATED":
+			return {
+				...state,
+				authenticated: false,
+				user: null,
+			};
+		case "ADD_TO_CART":
+			return {
+				...state,
+				cart: {
+					...state.cart,
+					[action.purhcaseType]: [
+						...state.cart[action.purhcaseType],
+						action.item,
+					],
+				},
+			};
+		case "REMOVE_FROM_CART":
+			return {
+				...state,
+				cart: {
+					...state.cart,
+					[action.purchaseType]: [],
+				},
+			};
+		case "LOG_OUT":
+			return initialState;
+		default:
```

Comment:
This should be **state**, not **initialState**
```suggestion
		case "LOG_OUT":
			return initialState;
		default:
```

Reply:
I initially used "state" but when I did, it did not reset state to unauthenticated, so it never logged the user out even though the token has been deleted. It just returned the authenticated state.

Reply:
Ah yes I see. So shouldn't you just update the state with no user?

Reply:
I am also trying to empty the user's cart and change their authentication to false, so basically I am resetting the entire state to the initial state.